rbc/remotejit.py

```diff
@@ -404,7 +411,7 @@ def bar(a, b):
     typesystem_aliases = dict()
 
     def __init__(self, host='localhost', port=11532,
-                 local=False, debug=False):
+                 local=False, debug=False, use_debug_allocator=False):
```

Comment:
What about use `debug=True` to enable the debug allocator?

Reply:
I think it's cleaner to have a separate flag for the debug allocator, especially because it works only with `local=True`: what should it do with `debug=True, local=False`? Complain or ignore the debug allocator? Better to be explicit, IMHO.

Also it seems that `debug` is a weird flag in `rbc`. E.g. it seems to me that it only increases verbosity but doesn't actually affect the generated code.

Reply:
"use_debug_allocator" contains two verbs "use" and "debug". Suggest using "debug_allocator".

Reply:
uhm, this is more an english detail than a technical one: in this sentence, "debug" is an adjective, like when we say e.g. "the debug version" or so. To it's meaning is "use the debug version of the allocator", at least as I intended it.
That say, I can change it if you want

Reply:
yes, this is English. According to google search, "debug" is most commonly a verb (can also be a noun).
On the other hand, I see what you mean: you have introduced `debug_allocator` module and intend to say to use this module. However, in this context, I think the existence of this module is irrelevant as an internal implementation detail and for users, the intent of the flag is "to debug the memory allocation/deallocation consistency".
If this makes sense, let's use "debug_allocator".

Reply:
I started to rename `use_debug_allocator` into `debug_allocator` but I quickly encountered a stylistic problem:
```python
from .rbclib import debug_allocator
[...]
    @dispatchermethod
    def call(self, fullname: str, arguments: tuple) -> Data:
        [...]
        if self.debug_allocator:
            leak_detector = debug_allocator.new_leak_detector()
```

i.e., now the `debug_allocator` name is overloaded with two meaning: the module itself and a bool indicating whether to use it or now. I think the above code is confusing, because it looks similiar-but-not-quite to a standard python pattern, which IMHO is confusing:
```python
if self.debug_allocator:
    leak_detector = self.debug_allocator.new_leak_detector()
```

no, now I am -0.5 to rename it

Reply:
> Rename the module?

good idea. I renamed it as `tracing_allocator`, which is a much better name. Thanks for convincing me to do this :)